### DISCUSSION TITLE
[Feature] Remove from Album in photo page

### DISCUSSION THREAD

**Discussion by mmomjian** [immich-app/immich#8390] [Feature Request]
### The feature

I have run into situations where I find an image in the timeline or by searching, and want to change what albums it is a part of. As far as I can tell, it is not possible to remove from album without being in the album itself, which means I have to find the picture in the album by scrolling.

Solution: Provide a "X" button next to the "Appears In" entries on the sidebar, to allow for easy removal of assets from an album.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by wardwygaerts** [comment]
I have the same request, I mis this feature when I'm browsing in my albums and want to remove an asset from this album.
For me this option can also be in the drop down menu behind the 3 dots, the same as when you are in the album itself and select a asset.
![image](https://github.com/immich-app/immich/assets/7568498/c5fb7155-0321-4627-ac8e-3df3b692116a)

**Comment by sirweazel** [comment]
![Screenshot_20240809-201146](https://github.com/user-attachments/assets/90f896a4-ef6b-49b0-8e17-7fb5c768137b)
Maybe I'm missing it. I created an album on my photos from the app (Android). But I wanted to remove a few photos from the album. The only option I find and use actually deletes the photo and puts it in the trash, and removes it from everywhere. I'm looking for a feature just to remove from one album. Is this related to the the issue you reported, am I missing it, or should I create a new issue?

I provided a screenshot from the app. When I hit remove from album, It removed the photo from the general photos view, And I had to restore it from the trash.

Edit: after exploring the app UI some more. In one screen, it does remove from the album and in the other it deletes the photo (remove asset).  So I think this is a bug. I'll create a new issue when I'm at a computer with my examples. 
![Screenshot_20240809-205659](https://github.com/user-attachments/assets/1010db68-b984-4c13-8a64-340dc0d3c538)

So both screen shots indicate same function, but have different results.

**Reply by alextran1502** [reply]
@sirweazel the issue has been fixed and will be available in the next release

**Comment by Mannshoch** [comment]
The Problem still exist. Could it be this fix never got released?

**Reply by nalsai** [reply]
It is available on the iOS app as expected, but on the website I always have to exit the preview and select the image to remove it from an album.

**Comment by SpongyBob1958** [comment]
It is frustrating to be going through photos in an album and then wanting to remove a photo from the album, only having to pop up to the album level and find the photo to remove. Since in the info section you show the albums the photos in, can't you have a little x mark or something that just takes it out of the album?

**Comment by happychriss** [comment]
https://github.com/immich-app/immich/discussions/22025
was closed as a duplicate  - I am aligned :-) Sorry for this

Just wanted to point out, that its not only relevant for the web but also for the mobile app.


I organize photos into albums while browsing in full-screen (not going back to thumbnails) - e.g. after a nice vacation day :-)

Current Situation:

Using version 1.142 on web and mobile:
- Web app: I can add a photo while browsing in fullscreen to an album, but I can not remove a photo from album, I have to go back to thumb-view.
- Mobile app: I can not add and not remove a photo while browsing in fullscreen. I have to go back to thumb-view for removal.

Feature requested
- Web app: I am also able to remove a photo from an album in fullscreen
- Mobile app: I am able to add and remove a photo from/to an album in fullscreen

I would give priority on the mobile app, because that is a more general use-case to work on the photos while e.g. travelling.

It would be great to have a one-click way to add/remove photos while in full-screen working for mobile and web-app.

**Reply by nalsai** [reply]
What version of the mobile app are you running? Because on my iPhone and my iPad I can do everything from both the fullscreen and the thumbnail view. It used to be where delete was when outside of an album but now is the right-most entry after tapping on the three dots (as seen in the screenshot on the second comment).

**Reply by happychriss** [reply]
Hi, sorry I am running Android for so long time already -forgot to mention:
Mobile - App: Android 1.142.0 build.3014
I test two scenarios:
a) Full screen view of a photo selected from photos-stream -> would expect option to add to album
b) Full screen view of a photo selected from album-view -> would expect option to remove from album

In both scenarios:
If I click on the 3 dots on top left of swipe upwards - I see below, this does not offer an option to add to album:
<img width="804" height="1750" alt="Screenshot From 2025-09-16 21-50-52" src="https://github.com/user-attachments/assets/8262500f-1ff0-4d35-81c9-d26cde2c4fd1" />

**Reply by nalsai** [reply]
Ah, sorry, I misunderstood your issue. Add to album is indeed missing from the full screen view on the app.

**Reply by lao9s** [reply]
I used to use this feature every day. After the last update, I don't have it anymore, so sad. I hope it gets restored.

**Comment by Escapist6898** [comment]
I agree with this feature request, assuming it is still open.

Version: 2.4.1, Android/Web

Currently, removing a photo from one or more albums is not intuitive, and in the case of the web, not easy.

1) On Android, it is possible by going to the 3-dots menu and choosing 'Remove from album'. This is acceptable, but having it on the bottom toolbar would be even better. I'd put it in place of 'Delete', since IMO you are more likely to remove from album than delete when you are in an album. 'Delete' would be better suited to be there when looking at a photo not from an album. In addition, it would be nice to have a 'Remove from album(s)' button instead of the 'Remove from album' which then allows you to remove the photo from multiple albums if the photo is in more than one, instead of having to go to each album separately.

2) However on the Web, this is even more obscure. The only way I have found to remove a photo from an album is to be on the album page itself and select the photo (not click into it), then top right from the menu button choose 'Remove from album' This is so unintuitive! Firstly there is plenty of space on the UI that a common feature as this shouldn't be hidden in the menu, and secondly, it should be available as a button (or at the very least within the menu) when on the individual photo. Even better would be for it to be available also in the 'Info' panel where you could remove it easily from multiple albums at the same time!

<img width="1415" height="455" alt="11" src="https://github.com/user-attachments/assets/25547ca2-cf0b-43a7-896d-869154d62ed3" />
<img width="377" height="254" alt="11" src="https://github.com/user-attachments/assets/b02d82a8-20f6-4a1c-a6c8-47f75331c129" />

**Comment by Horrih** [comment]
I also agree with this feature request, the current workflow is quite unintuitive IMHO.

When browsing your album in the single photo view, you need 6 manual  actions to remove a single photo from the current album :
- Leave the single photo view since it lacks the option to remove from album (see #13732)
- Find the photo you wanted to remove in the timeline and select it
- Click the three dots
- Click remove from album
- Click confirm
- Click on the next photo to resume your browsing

Ideally a button like suggested by @Escapist6898  with a single letter keyboard shortcut would be great !

**Comment by bo0tzz** [comment]
#27000

**Comment by maekawatoshiki** [comment]
#27000 only adds "Remove from album" to the asset viewer menu when the asset is opened from within an album.

The original request here (and https://github.com/immich-app/immich/discussions/8390#discussioncomment-14236755) was an "x" button next to each album under "Appears in" in the info panel.
That is still not available as of v3.2.2.

**Comment by mmomjian** [comment]
Re opening, indeed this is still not available from the info panel
